Core/src/IfcGeometryConverter/GeomUtils.cpp

```diff
@@ -1219,11 +1219,300 @@ bool GeomUtils::checkMeshSet( const carve::mesh::MeshSet<3>* mesh_set,
+
+		if (n) {
+			std::vector<double>	myArray(polynomialConstantCnt * n + 1);
+			GeomUtils::recursiveMultiplicationTaylor(
+				1.,
+				polynomialConstants, polynomialConstantCnt,
+				myArray, 0,
+				0, n
+			);
+
+			double	value = 0., factor = 1.;
+			for (size_t k = 0; k < polynomialConstantCnt * n + 1; k++)
+			{
+				factor *= s;
+				value += myArray[k] * factor / (k + 1);
+			}
+
+			size_t	tB = n;
+			while (tB > 1)
+			{
+				value /= (double)tB;
+				tB--;
+			}
+
+			return	value;
+		}
+		else {
+			return	s;
+		}
+	}
+	/**********************************************************************************************/
+
+	/*!Calculate i-value of cosine in the Taylor Series, namely negative or positiv value
+	* \param[in]  i                      i-integral step
+	* \param[in]  polynomialConstants    polynomial constants for a polyomial term - a,b,c,d
+	* \param[in]  polynomialConstantCnt  type of polinomial term - cubic, linear
+	* \param[in]  s                      the length of the curve between two points 
+	*
+	* \return                            cosine value
+	*/
+	double	GeomUtils::integralTaylorSeriesCosExpansion( 
+		size_t	i,
+		std::vector<double> polynomialConstants,
+		size_t	polynomialConstantCnt,
+		double	s
+	)
+	{
+		double	value =
+			computeIntegralTaylorSeriesExpansionElement(
+				i * 2,
+				polynomialConstants, polynomialConstantCnt,
+				s
+			);
+
+		if (i % 2) {
+			return	-value;
+		}
+		else {
+			return	value;
+		}
+	}
+	/**********************************************************************************************/
+
+	/*!Calculate i-value of sine in the Taylor Series, namely negative or positiv value
+	* \param[in]  i                      i-integral step
+	* \param[in]  polynomialConstants    polynomial constants for a polyomial term - a,b,c,d
+	* \param[in]  polynomialConstantCnt  type of polinomial term - cubic, linear
+	* \param[in]  s                      the length of the curve between two points
+	*
+	* \return                            sine value
+	*/
+	double	GeomUtils::integralTaylorSeriesSinExpansion(
+		size_t	i,
+		std::vector<double> polynomialConstants,
+		size_t	polynomialConstantCnt,
+		double	s
+	)
+	{
+		double	value =
+			computeIntegralTaylorSeriesExpansionElement(
+				i * 2 + 1,
+				polynomialConstants, polynomialConstantCnt,
+				s
+			);
+
+		if (i % 2) {
+			return	-value;
+		}
+		else {
+			return	value;
+		}
+	}
+	/**********************************************************************************************/
+
+	/*!Calculate cosinus for X in th Taylor Series
+	* \param[in]  polynomialConstants    polynomial constants for a polyomial term - a,b,c,d
+	* \param[in]  polynomialConstantCnt  type of polinomial term - cubic, linear
+	* \param[in]  s                      curve length betwee two points
+	*
+	* \return                            the value of cosine
+	*/
+	double	GeomUtils::integralTaylorSeriesCos(
+		std::vector<double> polynomialConstants,
```

Comment:
generally, if you're not changing the contents of a `std::vector` in a function always pass it as const reference:  `const std::vector<...>& name`

Reply:
Resolve 5442d2d